scripts/_aggregate_cells.py

```diff
@@ -0,0 +1,113 @@
+import os
+import argparse 
+from pathlib import Path
+
+import pandas as pd
+from pycytominer.cyto_utils.cells import SingleCells
+
+
+def aggregate_cells(plate, sql_file, metadata_dir, barcode_platemap, strata, image_cols, compression):
+    """ Aggregates cellular image data. Aggregates are convereted into compressed csv files.
+
+    Parameters
+    ----------
+    plate : str
+       plate identifier 
+    sql_file : str
+        path to file containing cell image data
+    metadata_dir : str
+        path to cell metadata diretory
+    barcode_platemap : str
+        path to file containing barcode metadata
+    strata : list[StringType]
+        columns used for grouping aggregate single cells
+    image_cols : list[StringTypes]
+        selected columns from cell image data 
+    compression : str
+        compression algorithm applied on aggregate data
+        
+    Raises
+    ------
+    FileNotFoundError
+        raised if the single cell data file
+    """
+    print("Processing {}.....".format(plate))
+
+    if not os.path.isfile(sql_file):
+        raise FileNotFoundError("Unable to find SQL file")
+
+    sqlite_file = "sqlite:///{}.sqlite".format(plate)
+
+    # Load specific platemap
+    barcode_platemap_df = pd.read_csv(barcode_platemap)
+    platemap = barcode_platemap_df.query(
+        "Assay_Plate_Barcode == @plate"
+    ).Plate_Map_Name.values[0]
+    platemap_file = os.path.join(metadata_dir, "platemap", "{}.csv".format(platemap))
+    platemap_df = pd.read_csv(platemap_file)
+
+    # Prepare sql file for processing
+    ap = SingleCells(
+        sqlite_file, strata=strata, image_cols=image_cols, fields_of_view_feature=[]
+    )
+
+    # Count cells and output
+    print("Counting cells")
+    results_path = Path("results")
+    results_path.mkdir(exist_ok=True)
+    cell_count_file = os.path.join("results", "{}_cell_count.tsv".format(plate))
+    cell_count_df = ap.count_cells()
+
+    print("Saving cell counts in: {}".format(cell_count_file))
+    cell_count_df = cell_count_df.merge(
+        platemap_df, left_on="Image_Metadata_Well", right_on="well_position"
+    ).drop(["WellRow", "WellCol", "well_position"], axis="columns")
+    cell_count_df.to_csv(cell_count_file, sep="\t", index=False)
+
+    # Begin processing profiles
+    print("Compressing profiles...")
+    data_path = Path("data")
+    data_path.mkdir(exist_ok=True)
+    output_dir = os.path.join("data", "profiles", plate)
+    os.makedirs(output_dir, exist_ok=True)
+
+    # Aggregate single cells into well profiles
+    out_file = os.path.join(output_dir, "{}.csv.gz".format(plate))
+    ap.aggregate_profiles(output_file=out_file, compression_options="gzip")
+    print("compressed file written on:")
+
+
+if __name__ == "__main__":
+
+    # CLI Arguments
```

Comment:
Oh interesting! Does snakemake require CLI args? How is this handled differently than the config file?

Reply:
`_aggregate_cells.py` Removed on commit: 67828fb3af04336c529db4eebfd2c0f8f3958e04

To answer you question. Snakelike handles CLI arguments under the `shell` rule parameter and must be explicitly written. Example below:

```python 
# example_rule2.smk
rule handle_cli:
    input:
        "/path/to/input"
    output:
        "/path/to/output"
    params:
        x="param1"
        y="param2"
        z="param3"
    shell: # uses shell rule parameter
        "python aggregate_cells.py -i {input} -o {output} -x {param1} -y {param2} -z {param3}"
```

as you can see, it can become problematic if more parameters are required for the script to execute. 

The new `aggregate_cells.py` script is called by using the `script` rule attribute. The `script` attribute allows `snakemake` to "magically inject"  parameters written in the `snakemake` file into the called script; therefore greatly reducing the length of the command. Example below: 

```python
# example_rule2.smk
rule handle_cli:
    input:
        "/path/to/input"
    output:
        "/path/to/output"
    params:
        x="param1"
        y="param2"
        z="param3"
    script: # replace `shell` with `script` reduces expressiveness of command string
        "aggregate_cells.py "
```

Accessibility of these parameters are simple! snakemake does a great job in keeping it the `pythonic` to access attributes via dot notation and hashing (dictionaries) 

```python
input = snakemake.input
param_x = snakemake.params["x"] # -> "param1"
param_y = snakemake.params["y"] # -> "param2"
param_z = snakemake.params["z"] # -> "param3"
```

What really nice about the `params` rule attribute is that it can take configurational files as inputs. Therefore, this will require the user to only interact with the config file and not the `snakemake` file

## Other notes
- `shell` is best to execute third party compiled programs. 
- `script` rule attribute is smart. It can differentiate which interpreter to execute based on the extension of the script. 

Hopefully this clears any confusion!